store/sql/src/main/java/org/apache/carbondata/dis/DisProducer.java

```diff
@@ -0,0 +1,151 @@
+      return;
+    }
+
+    DIS dic = DISClientBuilder.standard().withEndpoint(args[1]).withAk(args[3]).withSk(args[4])
+        .withProjectId(args[5]).withRegion(args[2]).build();
+
+    Sensor sensor = new Sensor(dic, args[0]);
+    Timer timer = new Timer();
+    timer.schedule(sensor, 0, 5000);
+
+  }
+
+  static class Sensor extends TimerTask {
+    private DIS dic;
+
+    private String streamName;
+
+    private SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+
+    private Random random = new Random();
+
+    private int i = 0;
+    private int flag = 1;
+
+    Sensor(DIS dic, String streamName) {
+      this.dic = dic;
+      this.streamName = streamName;
+    }
+
+    @Override public void run() {
+      uploadData();
+      //recordSensor();
+    }
+
+    private void uploadData() {
+      PutRecordsRequest putRecordsRequest = new PutRecordsRequest();
+      putRecordsRequest.setStreamName(streamName);
+      List<PutRecordsRequestEntry> putRecordsRequestEntryList = new ArrayList<>();
+      PutRecordsRequestEntry putRecordsRequestEntry = new PutRecordsRequestEntry();
+      putRecordsRequestEntry.setData(ByteBuffer.wrap(recordSensor()));
+      putRecordsRequestEntry
+          .setPartitionKey(String.valueOf(ThreadLocalRandom.current().nextInt(1000000)));
+      putRecordsRequestEntryList.add(putRecordsRequestEntry);
+      putRecordsRequest.setRecords(putRecordsRequestEntryList);
+
+      LOGGER.info("========== BEGIN PUT ============");
+
+      PutRecordsResult putRecordsResult = null;
+      try {
+        putRecordsResult = dic.putRecords(putRecordsRequest);
+      } catch (DISClientException e) {
+        LOGGER.error(e,
+            "Failed to get a normal response, please check params and retry." + e.getMessage());
+      } catch (ResourceAccessException e) {
+        LOGGER.error(e, "Failed to access endpoint. " + e.getMessage());
+      } catch (Exception e) {
+        LOGGER.error(e, e.getMessage());
+      }
+
+      if (putRecordsResult != null) {
+        LOGGER.info("Put " + putRecordsResult.getRecords().size() + " records[" + (
+            putRecordsResult.getRecords().size() - putRecordsResult.getFailedRecordCount().get())
+            + " successful / " + putRecordsResult.getFailedRecordCount() + " failed].");
+
+        for (int j = 0; j < putRecordsResult.getRecords().size(); j++) {
+          PutRecordsResultEntry putRecordsRequestEntry1 = putRecordsResult.getRecords().get(j);
+          if (putRecordsRequestEntry1.getErrorCode() != null) {
+            LOGGER.error("[" + new String(
+                putRecordsRequestEntryList.get(j).getData().array(), Charset.defaultCharset())
+                + "] put failed, errorCode [" + putRecordsRequestEntry1.getErrorCode()
+                + "], errorMessage [" + putRecordsRequestEntry1.getErrorMessage() + "]");
+          } else {
+            LOGGER.info("[" + new String(
```

Comment:
why not use stringbuilder or stringformat

Reply:
ok